### PR TITLE
Add first run welcome screen

### PR DESCRIPTION
**Pull request by NathanW2**
## Description
Super early preview so don't stress too much over how it works more just what it is.

This is a Welcome To QGIS X screen that can be shown on first run of the application.  It's shown before anything is setup in order to be used to guide users though any setup needed by the application, show what's new, etc. 

welcome

Also, don't freak out about the Usage Stats thing is just showing a use case for this kind of screen.

This also uses QML so it looks a bit nicer.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by pcav** [comment]
I like that, thanks!

File: CMakeLists.txt [hunk 1]
```diff
@@ -264,6 +264,7 @@ FIND_PACKAGE(Qt5Test REQUIRED)
 FIND_PACKAGE(Qt5UiTools REQUIRED)
 FIND_PACKAGE(Qt5Script REQUIRED)
 FIND_PACKAGE(Qt5Sql REQUIRED)
+FIND_PACKAGE(Qt5Qml REQUIRED)
```
Comment (m-kuhn) [inline CMakeLists.txt]: Can you move that over to app?

Comment (NathanW2) [inline CMakeLists.txt]: Yep

**Comment by elpaso** [comment]
Nice! Can you add hooks for plugins to customise?

**Comment by NathanW2** [comment]
So I'm going to keep working on this now that I have my new PC ;) 

After thinking about it for a while I'm not going to use QML for it at the moment. Qt Creator was using QML for theirs and it slowed down load times, etc.  So I I'm considering maybe HTML or just a plain QWidget for it.

**Comment by m-kuhn** [comment]
Would be great to bring this forward given #4994.

**Comment by NathanW2** [comment]
I'm working on a new welcome screen but that should not be used for tips
because it's only run once

**Comment by m-kuhn** [comment]
Oh yeah. No tips on first run! Maybe a link to a get started page...

**Comment by m-kuhn** [comment]
@NathanW2 are you still planning to continue this work?

**Comment by NathanW2** [comment]
For 3.2 yes, not sure I can do it for 3.0 though.  I guess it's not really
a feature so maybe I can get it in.

**Comment by m-kuhn** [comment]
Just make it ready whenever you have time. We can then assess if it's too late / too intrusive for 3.0 and if yes postpone to 3.2.

But I think it would be a big win in terms of UX for people upgrading.

**Comment by DelazJ** [comment]
> But I think it would be a big win in terms of UX for people upgrading

Fully agree...

**Comment by NathanW2** [comment]
Already done in master.

**Pull request closed without merging**